### PR TITLE
App 11253 tokens via api

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- [Ticket](https://flipapp.myjetbrains.com/issue/APP-11253)

### General

- [ ] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--page).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [ ] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--page).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--page).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--page).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--page).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--page)

### CODE REVIEW

**Review by Sqrrl — COMMENTED** [review]
Is it /api/v0/tokens or /api/tokens/v0? It gets mixed up in the Notion page.

File: apps/swirl-docs/src/pages/api/tokens/v0/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import type { NextApiRequest, NextApiResponse } from "next";
+import { OVERVIEW_HTML } from "./utils";
+
+export default function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<any>
+) {
+  res.setHeader("Content-Type", "text/html");
+  res.status(200).send(OVERVIEW_HTML);
+}
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api/tokens/v0/index.ts:11]: **suggestion:** Would be nice if all endpoints (including the overview) would respond with JSON as default.

File: apps/swirl-docs/src/pages/api/tokens/v0/utils.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { SWIRL_TOKENS_DIST_PATH } from "@swirl/lib/navigation";
+import fs from "fs";
+import path from "path";
+
+export const OVERVIEW_HTML = fs.readFileSync(
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api/tokens/v0/utils.ts:5]: **thought:** All readFile operation could be wrapped in a function, so the files get loaded into memory on demand. Tradeoff would be slightly longer response times.

File: packages/swirl-tokens/scripts/utils.ts [hunk 3]
```diff
@@ -125,6 +127,8 @@ export function createSwirlTailwindTheme() {
       oneTheme[key] = {};
     }
 
+    console.log("key", key);
```
Comment (Sqrrl) [inline packages/swirl-tokens/scripts/utils.ts:130]: **chore:** remove

File: apps/swirl-docs/src/pages/api/tokens/v0/utils.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+export type TokenGroupType =
+  | "colors"
+  | "spacing"
+  | "typography"
+  | "border"
+  | "zindex"
+  | "all";
+
+export const Formats = ["json", "css", "scss", "less", "flutter"];
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api/tokens/v0/utils.ts:45]: **nitpick:** "Formats" is a bit weird for a constant name. "formats" or "FORMATS"? An enum would be fine too.

File: apps/swirl-docs/src/pages/api/tokens/v0/[tokengroup].ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      case TokenFormats.CSS:
+        res.setHeader("Content-Type", "text/css");
+        if (scheme === "dark") {
+          res.status(200).send(cssDark);
+        } else {
+          res.status(200).send(cssLight);
+        }
+        break;
+      case TokenFormats.FLUTTER:
+        res.setHeader("Content-Type", "text/dart");
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api/tokens/v0/[tokengroup].ts:68]: **question:** Are text/dart, text/less and text/scss accepted mime types? If not, I think text/plain or application/octet-stream would be the safest option.

**Review by Sqrrl — CHANGES_REQUESTED** [review]

**Comment by Sqrrl** [comment]
There a console errors, eg. when calling `/api/tokens/v0/colors`.

File: apps/swirl-docs/src/pages/api/tokens/v0/index.ts (hunk 1 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/pages/api/tokens/v0/index.ts:11]: would agree, but other requirements were [set](https://flipapp.myjetbrains.com/youtrack/issue/APP-11253/Tokens-via-API#focus=Comments-4-21567.0-0) ^-^

**Comment by webspaceadam** [comment]
> Is it /api/v0/tokens or /api/tokens/v0? It gets mixed up in the Notion page.

so i used <code>/api/tokens/v0/[tokengroup]</code>

**Comment by Sqrrl** [comment]
> > Is it /api/v0/tokens or /api/tokens/v0? It gets mixed up in the Notion page.
> 
> i decided to use `/api/tokens/v0/[tokengroup]`

So the v0 refers to the token version, not the api version?

**Comment by webspaceadam** [comment]
> So the v0 refers to the token version, not the api version? I'm confused. What was @Jan10's idea behind this?

yes thats how i understood the notion page :)

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**